### PR TITLE
ocr3 - Add commit plugin e2e unit tests

### PR DESCRIPTION
**Pull request by dimkouv**
Add e2e unit tests using the `ocr3runner` library.
Includes minor refactoring and fixes.

### CODE REVIEW

File: core/services/ocr3/plugins/ccip/commit/plugin.go [hunk 1]
```diff
@@ -205,23 +206,18 @@ func (p *Plugin) Outcome(_ ocr3types.OutcomeContext, _ types.Query, aos []types.
...
 
-	maxSeqNums, err := maxSeqNumsConsensus(p.lggr, fChainDest, decodedObservations)
-	if err != nil {
-		return ocr3types.Outcome{}, fmt.Errorf("max sequence numbers consensus: %w", err)
-	}
-	if len(maxSeqNums) == 0 {
-		p.lggr.Warnw("no consensus on max sequence numbers")
-		return ocr3types.Outcome{}, nil
-	}
+	maxSeqNums := maxSeqNumsConsensus(p.lggr, fChainDest, decodedObservations)
```
Comment (makramkd) [inline core/services/ocr3/plugins/ccip/commit/plugin.go:220]: So its a valid scenario for `maxSeqNums` to be empty? What does that mean qualitatively in this scenario?

Comment (makramkd) [reply core/services/ocr3/plugins/ccip/commit/plugin.go:220]: Shouldn't there always be a max seq nr for each chain being observed?

Comment (makramkd) [reply core/services/ocr3/plugins/ccip/commit/plugin.go:220]: Ah I see, this is a "nil state" where there are no source chains?

File: core/services/ocr3/plugins/ccip/internal/mocks/reportcodec.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package mocks
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/smartcontractkit/ccipocr3/internal/model"
+)
+
+type CommitPluginJSONReportCodec struct{}
```
Comment (makramkd) [inline core/services/ocr3/plugins/ccip/internal/mocks/reportcodec.go:10]: Maybe add a comment that this is only valid in unit tests?

**Review by makramkd — APPROVED** [review]

File: core/services/ocr3/plugins/ccip/internal/mocks/reportcodec.go (hunk 2 above)
Comment (dimkouv) [reply core/services/ocr3/plugins/ccip/internal/mocks/reportcodec.go:10]: it's under mocks package, should be clear

**Pull request merged**